### DISCUSSION TITLE
Missing default return value for typer.Option

### DISCUSSION THREAD

**Discussion by wu-clan** [fastapi/typer#574] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Optional

import typer


def main(
        name: Optional[str] = typer.Option('Test', '--name', '-n'),
):
    if name:
        print(f"Good day Ms. {name}.")
    else:
        print(f"Hello World!")


if __name__ == "__main__":
    typer.run(main)
```

### Description

When executing `python scratch.py --name` or `python scratch.py -n`, happend `Option '-n' requires an argument.`, but it should detect the default value and return it instead of forcing a value to be filled in, if using `typer.Argument `, it doesn't have any problems

### Operating System

Windows

### Typer Version

0.6.1

### Python Version

3.8.10

**Comment by YuriiMotov** [comment, marked as answer]
The idea of default value is that it will be used if user doesn't specify this option at all
```
python scratch.py
```

`python scratch.py -n` will not work here, because Typer needs to know the exact number of values to expect for this option.
Otherwise we would have ambiguity: is the value after `-n` the value of this option or the value of another argument?
